app/controllers/v1/sessions_controller.rb

```diff
@@ -181,24 +181,11 @@ def saml_cookie_content
     # rubocop:disable Metrics/CyclomaticComplexity
     def login_params(type)
       raise Common::Exceptions::RoutingError, type unless REDIRECT_URLS.include?(type)
-
-      case type
-      when 'signup'
-        url_service.signup_url
-      when 'mhv'
-        url_service.mhv_url
-      when 'dslogon'
-        url_service.dslogon_url
-      when 'idme'
-        url_service.idme_url
-      when 'mfa'
-        url_service.mfa_url
-      when 'verify'
-        url_service.verify_url
-      when 'custom'
+      if type == 'custom'
         raise Common::Exceptions::ParameterMissing, 'authn' if params[:authn].blank?
-
         url_service(false).custom_url params[:authn]
+      else
+        url_service.send("#{type}_url")
```

Comment:
```suggestion
        url_service.send("#{type}_url") if url_service.respond_to?("#{type}_url")
```

Reply:
Funny, I was thinking the same thing after digging deeper, this just doesn't belong here. Maybe I'll take a look at the PostURLService. I think I was just looking in this one file. Something like: ```url_service.url_for_type(type)```

Reply:
Yeah, I feel like that approach is super straightforward. I just now did a quick check, and I think you'd only have to refactor the functions in both sessions controllers (`v0` and `v1`), so maybe wouldn't be the worst refactor ever